crates/shared/src/trade_finding/external.rs

```diff
@@ -206,13 +206,35 @@ mod dto {
     }
 
     #[serde_as]
-    #[derive(Clone, Debug, Deserialize)]
+    #[derive(Clone, Debug)]
     pub enum QuoteKind {
         Legacy(LegacyQuote),
         #[allow(unused)]
         Regular(Quote),
     }
 
+    impl<'de> Deserialize<'de> for QuoteKind {
```

Comment:
Why re-invent the wheel here? Isn't it more straight forward to define the enum with 2 variants, let serde auto generate the deserialization code and produce the error when converting from `QuoteKind` to `Trade` if `QuoteKind::Regular`?

Reply:
Looks like I just had to use `#[serde(untagged)]` to make it work. Missed that initially.